github/repos_releases.go

```diff
@@ -211,6 +213,54 @@ func (s *RepositoriesService) GetReleaseAsset(owner, repo string, id int) (*Rele
 	return asset, resp, err
 }
 
+// DowloadReleaseAsset downloads a release asset.
+//
+// DowloadReleaseAsset returns an io.ReadCloser that reads the contents of the
+// specified release asset. It is the caller's responsibility to close the ReadCloser.
+//
+// GitHub API docs : http://developer.github.com/v3/repos/releases/#get-a-single-release-asset
+func (s *RepositoriesService) DowloadReleaseAsset(owner, repo string, id int) (io.ReadCloser, error) {
+	u := fmt.Sprintf("repos/%s/%s/releases/assets/%d", owner, repo, id)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Accept", defaultMediaType)
+
+	var resp *http.Response
+	if s.client.client.Transport == nil {
+		resp, err = http.DefaultTransport.RoundTrip(req)
+	} else {
+		resp, err = s.client.client.Transport.RoundTrip(req)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// GitHub API streamed the asset directly
```

Comment:
Go's default HTTP client will transparently follow redirects (up to 10 hops), so you don't need to differentiate between streamed directly and redirected assets... from the clients perspective, they will all look like they are streamed directly.  Instead of digging into the `s.client.client.Transport` above, just call `s.client.client.Get` as is done in [DownloadContents](https://github.com/google/go-github/blob/9420d0f48acb838ccf8a7f8dde9eaee19af71132/github/repos_contents.go#L112).